File: top100/No49_userReadAuthority.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hashmap;
        int length = strs.size();
        for (int i =0; i < length; ++i) {
            char cKey[] = vector<i>.c_str();
            std::sort(cKey, 0,cKey.size());
            string key = cKey;
            vector<string>::iterator it = find(hashmap.begin(), hashmap.end(), key);
            if (it == hashmap.end()) {
                hashmap.push_back(key, new vector<string>());
            }

        }
        string iter = 
    }
};

File: top100/No49_userReadAuthority.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if (strs.length == 0) return new ArrayList(); 
        Map<String, List> ans = new HashMap<String, List>();
        for(String str : strs) {
            char[] ca = str.toCharArray();
            Arrays.sort(ca);
            String key = String.valueOf(ca);
            System.out.println(key);
            if (!ans.containsKey(key)) {
                ans.put(key, new ArrayList());    
            }
            ans.get(key).add(str);
        }
        return new ArrayList(ans.values());
    }
} 